relayer/crates/starknet-chain-components/src/impls/starknet_to_cosmos/consensus_state.rs

```diff
@@ -0,0 +1,24 @@
+use hermes_core::chain_components::traits::{
+    ConsensusStateFieldComponent, ConsensusStateFieldGetter, HasConsensusStateType, HasTimeType,
+};
+use hermes_prelude::*;
+use tendermint::Time;
+
+use crate::types::CometConsensusState;
+
+pub struct ProvideCometConsensusState;
+
+#[cgp_provider(ConsensusStateFieldComponent)]
+impl<Chain, Counterparty> ConsensusStateFieldGetter<Chain, Counterparty>
+    for ProvideCometConsensusState
+where
+    Chain: HasConsensusStateType<Counterparty, ConsensusState = CometConsensusState>,
+    Counterparty: HasTimeType<Time = Time>,
+{
+    fn consensus_state_timestamp(consensus_state: &CometConsensusState) -> Counterparty::Time {
+        let timestamp = consensus_state.timestamp;
+        let (secs, nanos) = (timestamp / 1_000_000_000, timestamp % 1_000_000_000);
```

Comment:
you may use a constant like `NANOS_PER_SEC` to avoid typos here.

Reply:
Done in 88d70446de1184817d535f753ddcb99075a0d5c9